enumerables.rb

```diff
@@ -0,0 +1,121 @@
+module Enumerable
+    def my_each
+        if block_given?
+            item = 0
+            while item < self.size
+                yield(self[item])
+             item+=1
+            end
+        end
+    
+    self
+    end
+
+
+
+    def my_each_with_index
+        if block_given?
+            index = 0
+            while index < self.size
+                yield(self[index], index)
+                index+=1
+            end
+        end
+        self
+    end
+
+    def my_select
+        if block_given?
+            filtered = []
+            index = 0
+            self.my_each { |x| filtered.push(x) if yield(x) }
+        end
+        filtered
+    end
+    
+    def my_all?
+        if block_given?
+            not_all = 0
+            index = 0
+            while index < self.size
+                not_all = 1 if !yield(self[index])
+                index+=1
+                break if not_all == 1
+            end
+            
+        end
+        not_all == 1 ? false : true
+    end
```

Comment:
`my_any?`, `my_none?` and `my_all?` should be able to accept patterns which check each member of the collection.

> If instead a pattern is supplied, the method returns whether pattern === element for every collection member.

```ruby
   %w[ant bear cat].all?(/t/) #=> false
   %w[ant bear cat].my_all?(/t/) #=> ArgumentError (wrong number of arguments (given 1, expected 0))
```

Please see other [examples of `all` from the ruby doc](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F)